runtime/pkg/duckdbsql/expression.go

```diff
@@ -0,0 +1,38 @@
+package duckdbsql
+
+import (
+	"errors"
+	"fmt"
+)
+
+// EvaluateBool uses DuckDB to evaluate the given expression as a boolean value.
+func EvaluateBool(expr string) (bool, error) {
+	// NOTE: Doing type check in SQL instead of in Go to protect against expressions that return huge values.
+	res, isBool, err := queryBoolPair(fmt.Sprintf("SELECT v::BOOLEAN AS b, TYPEOF(v) = 'BOOLEAN' AS is_bool FROM (SELECT (%s) AS v)", expr))
```

Comment:
`v::BOOLEAN` will try to cast v as boolean so if its not boolean, query will fail so whats the need of `is_bool` expression. We can just rely on err from query to check if its boolean or not ?

Reply:
ok was not aware of this, so this looks good.